src/client/app/core/api/routes.service.js

```diff
@@ -7,31 +7,46 @@
   routes.$inject = ['$http', 'user'];
 
   function routes ($http, user) {
-    var _this = this;
-    _this.routes = [];
+    var vm = this;
+    vm = {
+      routes: [],
+      addRoute: addRoute,
+      updateRoute: updateRoute,
+      fetch: fetch
+    };
+
+    return vm; 
+
+    function deleteRoute(){}
 
-    this.deleteRoute = function (){};
+    function addRoute(body, username){
+      console.log(body);
+      console.log(vm);
+      var route = {
+        username: username,
+        route: body.route,
+        methods: {
 
-    this.addRoute = function(body){
+        },
+        headers: '',
+        body: {}
+      };
+      console.log('routes',vm.routes);
 
-      var route = {};
-      route.method = 'GET';
-      route.route = body;
-      route.responseStatus = 200;
-      this.routes.push(route);
+      vm.routes.push(body.route);
+      console.log('routes',vm.routes);
       return $http({
         method: 'POST',
-        url: '/api/users/' + user.username + '/endpoints',
+        url: '/api/users/' + username + '/endpoints',
         data: route
-      }).then(function(result) {
+      }).success(function(result) {
         console.log('ADD ROUTE SUCCESS:', result);
```

Comment:
please add explicit `return` statement to trigger the chaining `.then` statements.